admin/stats_sales_report_graphs.php

```diff
@@ -55,15 +55,16 @@
 
 // check start and end Date
 $startDate = "";
-if (($_GET['startDate']) && (zen_not_null($_GET['startDate']))) {
+if (!empty($_GET['startDate']) && $_GET['startDate'] >= '0001-01-01') {
   $startDate = $_GET['startDate'];
 }
 $endDate = "";
-if (($_GET['endDate']) && (zen_not_null($_GET['endDate']))) {
+if (!empty($_GET['endDate']) && $_GET['endDate'] >= '0001-01-01') {
   $endDate = $_GET['endDate'];
 }
 // check filters
-if (($_GET['filter']) && (zen_not_null($_GET['filter']))) {
+$sales_report_filter = '';
+if (isset($_GET['filter']) && $_GET['filter'] && zen_not_null($_GET['filter'])) {

```

Comment:
Sorry, thought I had responded to this, but in reviews see that I hadn't.

Seems a bit odd; however, now that I've seen this code in operation, I see that the `filter` is a string representation of binary on/off for each of the associated items to be displayed/filtered.  As a result, for some reason `($filter) == '0'` works (even `$filter == '0'` works) if all characters are 0; however, `empty($filter)` does not evaluate to `true` if all characters are 0.  So the substitution of `empty` for an evaluation of `== '0'` does not seem to resolve as "perhaps" expected.

I mean, it (say '0000') doesn't correlate to an empty characteristic identified for the `empty` "function"; however, I also didn't expect `'0000' == '0'` to be truthy either...  Weird, here is the result of some testing with the result presented to the right:
`if ('0000')` = true
`if ('0000' == true)` = true
`if ('0000' == false)` = false
`if ('0000' == 1)` = false
`if ('0000' == '1')` = false
`if ('0000' == 0)` = true
`if ('0000' == '0')` = true
`if (empty('0000'))` = false
`if (zen_not_null('0000'))` = true

So, changing line 67 to `!empty()` *would* result in execution of the sub-code being executed even when all values are '0000', but would result in using `SALES_REPORT_GRAPHS_FILTER_DEFAULT` on first load or when left off of the URI (page loaded with the parameter removed).

But on line 303, if `empty` were used, then the string would only be built onto when the available size was 1 or zero.   

And actually, looking at the flow and logic, seems that the test case of `== '0'` is wrong here anyways.  Seems that the goal is to test if the string length is equal to the `status_available_size` and if not then to add '0`s until it is. Something like:
````php
          if (strlen($sales_report_filter) < $report->status_available_size) {
            for ($i = strlen($sales_report_filter); $i < $report->status_available_size; $i++) {
              $sales_report_filter .= "0";
            }
          }
````
Which really could be shortened to just:
````php
          for ($i = strlen($sales_report_filter); $i < $report->status_available_size; $i++) {
            $sales_report_filter .= "0";
          }
````
Because the internal code will not be executed if `strlen($sales_report_filter) >= $report->status_available_size`.

Just some thoughts on this... 

Stepped in to try to eliminate some myDebug logs that I came across, but starting to seem like there is more to be corrected than just strict notifications.

Reply:
ya, there's room for future optimization, but this solves the strict-notices for now.